pkgs/development/compilers/yosys/default.nix

```
@@ -46,17 +43,18 @@ stdenv.mkDerivation rec {
   };
 
   enableParallelBuilding = true;
-  nativeBuildInputs = [ pkg-config protobuf flex bison python3 ];
-  buildInputs = [ tcl readline libffi python3 protobuf zlib ];
+  nativeBuildInputs = [ pkg-config ];
+  buildInputs = [ tcl readline libffi python3 bison flex protobuf zlib ];
```

```suggestion
  nativeBuildInputs = [ bison flex pkg-config ];
  buildInputs = [ tcl readline libffi python3 protobuf zlib ];
```

---

```suggestion
  nativeBuildInputs = [ pkg-config protobuf flex bison python3 ];
  buildInputs = [ tcl readline libffi python3 protobuf zlib ];
```

protobuf and python3 are required at build-time

---

`python3` and `protobuf` must be linked to in this case (they are used at runtime), so leaving them in `buildInputs` is fine. `flex` and `bison` have been moved, however.